controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/pkg/ZipStreamReader.java

```diff
@@ -59,6 +63,24 @@ public static void transferAndWrite(OutputStream out, InputStream in, String nam
         }
     }
 
+    public static SortedMap<String, Long> getEntryCRCs(InputStream in, Predicate<String> entryNameMatcher) {
+        SortedMap<String, Long> entryCRCs = new TreeMap<>();
+        byte[] buffer = new byte[2048];
+        try (ZipInputStream zipIn = new ZipInputStream(in);
+             OutputStream os = new ByteArrayOutputStream()) {
+            for (ZipEntry entry = zipIn.getNextEntry(); entry != null; entry = zipIn.getNextEntry()) {
+                if (!entryNameMatcher.test(entry.getName()))
+                    continue;
+                // CRC is not set until entry is read
```

Comment:
Hmm ... in that case, perhaps we should consider a different hash? I'm not sure which is the most appropriate, but we should try to avoid collisions. It may be that the CRC is OK for this purpose, of course.